prov/efa/src/rxr/rxr_op_entry.c

```diff
@@ -216,3 +216,43 @@ size_t rxr_tx_entry_max_req_data_capacity(struct rxr_ep *ep, struct rxr_op_entry
 						      header_flags,
 						      tx_entry->rma_iov_count);
 }
+
+/**
+ * @breif set the max_req_data_size field of a tx_entry for multi-req
+ *
+ * Multi-REQ protocols send multiple REQ packets via one call to ibv_post_send() for efficiency.
+ * Under such circumstance, it is better that the data size of mulitple REQ packets to be close.
+ *
+ * To achieve the closeness, the field of max_req_data_size was introduced to rxr_op_entry,
+ * and used to limit data size when construct REQ packet.
+ *
+ * This function set the max_req_data_size properly.
+ *
+ *
+ * @param[in]		ep		endpoint
+ * @param[in,out]	tx_entry	tx_entry that has all information of
+ * 					a send operation
+ * @param[in]		pkt_type	type of REQ packet
+ *
+ */
+void rxr_tx_entry_set_max_req_data_size(struct rxr_ep *ep, struct rxr_tx_entry *tx_entry, int pkt_type)
+{
+	int max_req_data_capacity;
+	int mulreq_total_data_size;
+	int num_req;
+	static int MEMORY_ALIGNMENT = 8;
+
+	assert(rxr_pkt_is_mulreq(pkt_type));
+
+	max_req_data_capacity = rxr_tx_entry_max_req_data_capacity(ep, tx_entry, pkt_type);
+	assert(max_req_data_capacity);
+
+	mulreq_total_data_size = rxr_op_entry_mulreq_total_data_size(tx_entry, pkt_type);
+	assert(mulreq_total_data_size);
+
+	num_req = (mulreq_total_data_size - 1)/max_req_data_capacity + 1;
```

Comment:
why is this calculation different from rxr_tx_entry_num_req. Shouldn't you just use that function here?

Reply:
No. That function uses `rxr_tx_entry->max_req_data_size`, which is set by this function.

At the very begining, we do not know how many packets we are  going to send. so we need to determine that by using data_capacity.

Once we know the number of packet, we can calculate the average packet size then round up to alignment. 

we then calculate the number of packet again, which is `rxr_op_entry_num_req` for